### PR TITLE
ROX-26663: Add additional retries to Delegated Scanning Tests

### PR DESCRIPTION
**Pull request by dcaravel**
### Description

Adds retries to various actions that have caused the delegated scanning e2e tests to fail. Specifically:
- Retry obtaining a reference to the Sensor pod if multiple Sensor pods are found
  - When registry mirrors are set in OCP, OCP will drain every pod from every node, this takes time.  To speed up Sensor reconnecting to Central the tests restart Sensor and then wait for the Central/Sensor connection to be HEALTHY.  In one of the observed failures the connection was ready but there were two Sensor pods running (presumable one in a 'terminating' state).  The retries should give the pod enough time to shutdown before proceeding with the tests.
- Similarly, after the Central/Sensor connection was HEALTHY (which is determined via an API request to Central) - a subsequent request to the Central API triggered a 'connection refused' error from Central/k8s.
  - Retries were added when this happens to give Central / k8s enough time to stabilize.

Also increases usage to the `logf` wrapper so that timestamps appear in test output, so that can correlate timestamps between test logs to pod logs.

It may help when reviewing to review each commit in order.

### User-facing documentation

- [x] CHANGELOG is updated **OR** update is not needed
- [x] [documentation PR](https://spaces.redhat.com/display/StackRox/Submitting+a+User+Documentation+Pull+Request) is created and is linked above **OR** is not needed

### Testing and quality

- [x] the change is production ready: the change is GA or otherwise the functionality is gated by a feature flag
- [x] CI results are inspected

#### Automated testing

- [x] modified existing tests

#### How I validated my change

CI

### CODE REVIEW

**Review by RTann — APPROVED** [review]

File: tests/delegated_scanning_test.go [hunk 1]
```diff
@@ -942,20 +919,40 @@ func (ts *DelegatedScanningSuite) resetConfig(ctx context.Context) {
...
+func (ts *DelegatedScanningSuite) updateConfigWithRetries(ctx context.Context, cfg *v1.DelegatedRegistryConfig) error {
+	t := ts.T()
+
+	conn := centralgrpc.GRPCConnectionToCentral(t)
+	service := v1.NewDelegatedRegistryConfigServiceClient(conn)
+
+	retryFunc := func() error {
+		_, err := service.UpdateConfig(ctx, cfg)
+
+		if err != nil && strings.Contains(err.Error(), "connection refused") {
```
Comment (porridge) [inline tests/delegated_scanning_test.go:946]: Can you please extract a function that looks for this string and mangles `err` and reuse it below?

**Review by porridge — APPROVED** [review]
❤️ 
just one nitpick inline

**Pull request merged**